neuro_flow/expr.py

```diff
@@ -703,9 +817,13 @@ async def eval(self, root: RootABC) -> Optional[_T]:
                 # TODO: add str() function, raise an explicit error if
                 # an expresion evaluates non-str type
                 # assert isinstance(val, str), repr(val)
-                ret.append(str(val))
+                ret.append(val)
             try:
-                return self.convert("".join(ret))
+                if self.type is str:
```

Comment:
As is see above implicit concatenation is allowed for `issubclass(self.type, (str, RemotePath, LocalPath, URL))`, but this check is more strict.

Reply:
Good catch!